asyncpg/connect_utils.py

```diff
@@ -179,14 +179,19 @@ def _parse_hostlist(hostlist, port):
         port = _validate_port_spec(hostspecs, port)
 
     for i, hostspec in enumerate(hostspecs):
-        addr, _, hostspec_port = hostspec.partition(':')
-        hosts.append(addr)
+        if hostspec.startswith('/'):
+            hosts.append(hostspec)
+            # keep sequence ordering
+            hostlist_ports.append(5432)
```

Comment:
This is wrong.  Socket paths still have to take the port into consideration.  Use `port or default_port[i]` here.

Reply:
Well `default_port` at that index might not exist, right?

Reply:
The length of the `default_port` list is guaranteed to be the same as the number of specified hosts.  See `_validate_port_spec`

Reply:
Simply replace

```python
addr, _, hostspec_port = hostspec.partition(':')
```

with

```python

if not hostspec.startswith('/'):
    addr, _, hostspec_port = hostspec.partition(':')
else:
    addr = hostspec
    hostspec_port = ''
```